cloudinit/net/dhcp.py

```diff
@@ -503,62 +540,254 @@ def get_latest_lease(lease_dir, lease_file_regex):
+                            "dhcpcd is still running"
+                        )
+                    else:
+                        return lease
+                    time.sleep(sleep_time)
+                LOG.debug(debug_msg)
+                return lease
+            raise NoDHCPLeaseError("No lease found")
+
+        except subp.ProcessExecutionError as error:
+            LOG.debug(
+                "dhclient exited with code: %s stderr: %r stdout: %r",
+                error.exit_code,
+                error.stderr,
+                error.stdout,
+            )
+            raise NoDHCPLeaseError from error
+
+    @staticmethod
+    def parse_dhcpcd_lease(lease_dump: str, interface: str) -> Dict:
+        """parse the output of dhcpcd --dump
+
+        map names to the datastructure we create from dhclient
+
+        example dhcpcd output:
+
+        broadcast_address='192.168.15.255'
+        dhcp_lease_time='3600'
+        dhcp_message_type='5'
+        dhcp_server_identifier='192.168.0.1'
+        domain_name='us-east-2.compute.internal'
+        domain_name_servers='192.168.0.2'
+        host_name='ip-192-168-0-212'
+        interface_mtu='9001'
+        ip_address='192.168.0.212'
+        network_number='192.168.0.0'
+        routers='192.168.0.1'
+        subnet_cidr='20'
+        subnet_mask='255.255.240.0'
+        """
+
+        # create a dict from dhcpcd dump output - remove single quotes
+        lease = dict(
+            [
+                a.split("=")
+                for a in lease_dump.strip().replace("'", "").split("\n")
+            ]
+        )
+
+        # this is expected by cloud-init's code
+        lease["interface"] = interface
+
+        # transform underscores to hyphens
+        lease = {key.replace("_", "-"): value for key, value in lease.items()}
+
+        # - isc-dhclient uses the key name "fixed-address" in place of
+        #   "ip-address", and in the codebase some code assumes that we can use
+        #   isc-dhclient's option names. Map accordingly
+        # - ephemeral.py we use an internal key name "static_routes" to map
+        #   what I think is some RHEL customization to the isc-dhclient
+        #   code, so we need to match this key for use there.
+        name_map = {
+            "ip-address": "fixed-address",
+            "classless-static-routes": "static_routes",
+        }
+        for source, destination in name_map.items():
+            if source in lease:
+                lease[destination] = lease.pop(source)
+        return lease
+
+    def get_newest_lease(self, interface: str) -> Dict[str, Any]:
+        """Return a dict of dhcp options.
+
+        @param interface: which interface to dump the lease from
+        @raises: InvalidDHCPLeaseFileError on empty or unparseable leasefile
+            content.
+        """
+        try:
+            return self.parse_dhcpcd_lease(
+                subp.subp(
+                    [
+                        "dhcpcd",
```

Comment:
For our --dumplease command we aren't providing the full path to /usr/sbin/dhcpcd as we are for the initial --waitip command. This could lead unexpected issues in the unlikely event that $PATH contains a different locally installed dhcpcd. Shouldn't we be invoking the full path `/usr/sbin/dhcpcd for all commands to avoid potential corner case behavior?

Reply:
oops, yes thank you